apps/charterafrica/src/pages/_error.page.js

```diff
@@ -0,0 +1,38 @@
+import Error from "@/charterafrica/components/Error";
+import {
+  fetchGlobalProps,
+  fetchPage,
+} from "@/charterafrica/lib/data/fetchData";
+
+export default function CustomError({ blocks }) {
+  return blocks?.map((block) => {
+    switch (block?.slug) {
+      case "500":
+        return <Error key={block.slug} {...block} />;
+      default:
+        return null;
+    }
+  });
+}
+// TODO find a way to deal with other status codes
+// Status code is only accessible on getStaticProps but does not work with payload
+export async function getStaticProps(args) {
```

Comment:
`getInitialProps` is missing crucial arguments such as `locale`  @kilemensi

Reply:
What I'm trying to bring to your attention @koechkevin is the fact that `_error` needs to be rendered whenever an error occurs; client-side, server-side, at runtime. This means `getStaticProps` **should not** be used i.e. if we've set `statusCode` to **500** and generate html file for that at build time (that's what `getStaticProps` does), what will happen when **501** error occurs?

As long as you've read the [caveats](https://nextjs.org/docs/advanced-features/custom-error-page#caveats), if `getServerSideProps` work, great lets use that. Otherwise, we'll have to use `getInitialProps`.... with a bit of planning ahead, we can forward `locale` information to `getInitialProps` through the [custom](https://github.com/vercel/next.js/discussions/20242#discussioncomment-232060) [app](https://stackoverflow.com/a/65438136).